scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -180,11 +185,25 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
         .flatMap(
             metadataDirectory ->
                 Stream.of(metadataDirectory.listFiles())
-                    .filter(file -> file.getName().endsWith(".metadata.xml")))
+                    .filter(
+                        file ->
+                            file.getName()
+                                .matches("^([A-Za-z0-9]+(_[A-Za-z0-9]+){3})\\.metadata\\.xml$")))
         .map(RunProcessor::parseXml)
         .filter(Optional::isPresent)
         .forEach(metadata -> processMetadata(metadata.get(), dto, tz));
 
+    // Check if we grabbed a run alias from metadata.xml
+    if (dto.getRunAlias() != null) {
+      // Not valid, if it doesn't match run directory name
+      if (!runDirectory.getName().equals(dto.getRunAlias())) {
+        dto.setRunAlias(null);
```

Comment:
This is an error that makes the run invalid, so throw a runtime exception that states that. Once we know a run is invalid, we should stop wasting time processing it.

Would also be good to leave comment about why this is necessary since it's not obvious - when the run is first started, we can only get the name from the directory, so we need to make sure the same run doesn't show up under a different name later when the metadata files are available.